# Project 01 -- Anthony Micco

**TA Help:** N/A
    
**Collaboration:** Used information from the Stack Overflow page (https://stackoverflow.com/questions/5761212/how-do-i-count-the-number-of-rows-and-columns-in-a-file-using-bash) to find the bash command to calculate the number of rows in a CSV and not count the header row. 

In [2]:
%%bash

echo hello world

hello world


## Question 1

In [1]:
#setting the working directory to the /anvil/projects/tdm/data/bay_area_bike_share/kaggle/ directory

In [3]:
%%bash 
cd /anvil/projects/tdm/data/bay_area_bike_share/kaggle/
ls
pwd

archive.zip
bay_area_bike_share.db
station.csv
status.csv
trip.csv
weather.csv
/anvil/projects/tdm/data/bay_area_bike_share/kaggle


In [4]:
#getting the number of columns in the status.csv file 

In [19]:
%%bash

head -n2 /anvil/projects/tdm/data/bay_area_bike_share/kaggle/status.csv

station_id,bikes_available,docks_available,time
2,2,25,2013/08/29 12:06:01


In [6]:
#getting the size of the status.csv file

In [7]:
%%bash
cd /anvil/projects/tdm/data/bay_area_bike_share/kaggle/
ls -la
pwd

total 5240353
drwxrwsr-x+ 2 nobody nogroup       4096 Jul 28  2023 .
drwxrwsr-x+ 4 nobody nogroup      16384 Jul 28  2023 ..
-rw-rw----  1 nobody nogroup  581418405 Jul 24  2023 archive.zip
-rw-r--r--  1 nobody nogroup 2712824832 Nov 14  2019 bay_area_bike_share.db
-rw-r--r--  1 nobody nogroup       5647 Nov 14  2019 station.csv
-rw-r--r--  1 nobody nogroup 1989696383 Nov 14  2019 status.csv
-rw-r--r--  1 nobody nogroup   80208848 Nov 14  2019 trip.csv
-rw-r--r--  1 nobody nogroup     438063 Nov 14  2019 weather.csv
/anvil/projects/tdm/data/bay_area_bike_share/kaggle


A) There are 4 columns in the dataset

B) The size of the status.csv file is 1989696383 MB.

## Question 2

In [8]:
#Getting the months and years bay_area_bike_share/baywheels directory

In [60]:
%%bash 
cd /anvil/projects/tdm/data/bay_area_bike_share/baywheels
ls
pwd

2017-fordgobike-tripdata.csv
201801-fordgobike-tripdata.csv
201802-fordgobike-tripdata.csv
201803-fordgobike-tripdata.csv
201804-fordgobike-tripdata.csv
201805-fordgobike-tripdata.csv
201806-fordgobike-tripdata.csv
201807-fordgobike-tripdata.csv
201808-fordgobike-tripdata.csv
201809-fordgobike-tripdata.csv
201810-fordgobike-tripdata.csv
201811-fordgobike-tripdata.csv
201812-fordgobike-tripdata.csv
201901-fordgobike-tripdata.csv
201902-fordgobike-tripdata.csv
201903-fordgobike-tripdata.csv
201904-fordgobike-tripdata.csv
201905-baywheels-tripdata.csv
201906-baywheels-tripdata.csv
201907-baywheels-tripdata.csv
201908-baywheels-tripdata.csv
201909-baywheels-tripdata.csv
201910-baywheels-tripdata.csv
201911-baywheels-tripdata.csv
201912-baywheels-tripdata.csv
202001-baywheels-tripdata.csv
202002-baywheels-tripdata.csv
202003-baywheels-tripdata.csv
202005-baywheels-tripdata.csv
202006-baywheels-tripdata.csv
202007-baywheels-tripdata.csv
202008-baywheels-tripdata.csv
202009-baywheels-tripdata

In [12]:
#getting the years represented in the years directory

In [15]:
%%bash 
cd /anvil/projects/tdm/data/election
ls
pwd

escaped2020sample.txt
itcont1980.txt
itcont1982.txt
itcont1984.txt
itcont1986.txt
itcont1988.txt
itcont1990.txt
itcont1992.txt
itcont1994.txt
itcont1996.txt
itcont1998.txt
itcont2000.txt
itcont2002.txt
itcont2004.txt
itcont2006.txt
itcont2008.txt
itcont2010.txt
itcont2012.txt
itcont2014.txt
itcont2016.txt
itcont2018.txt
itcont2020.txt
itcont2022.txt
itcont2024.txt
/anvil/projects/tdm/data/election


## Question 3

In [16]:
# testing example used in examples book

In [20]:
%%bash

head -n6 /anvil/projects/tdm/data/bay_area_bike_share/kaggle/trip.csv

id,duration,start_date,start_station_name,start_station_id,end_date,end_station_name,end_station_id,bike_id,subscription_type,zip_code
4576,63,8/29/2013 14:13,South Van Ness at Market,66,8/29/2013 14:14,South Van Ness at Market,66,520,Subscriber,94127
4607,70,8/29/2013 14:42,San Jose City Hall,10,8/29/2013 14:43,San Jose City Hall,10,661,Subscriber,95138
4130,71,8/29/2013 10:16,Mountain View City Hall,27,8/29/2013 10:17,Mountain View City Hall,27,48,Subscriber,97214
4251,77,8/29/2013 11:29,San Jose City Hall,10,8/29/2013 11:30,San Jose City Hall,10,26,Subscriber,95060
4299,83,8/29/2013 12:02,South Van Ness at Market,66,8/29/2013 12:04,Market at 10th,67,319,Subscriber,94103


In [21]:
#getting the starting and ending location and storing it in my home directory

In [22]:
%%bash

cut -d, -f4,7 /anvil/projects/tdm/data/bay_area_bike_share/kaggle/trip.csv >$HOME/startandendlocations.csv

In [23]:
#checking to ensure the data is there

In [38]:
%%bash
cd $HOME
head startandendlocations.csv
pwd

start_station_name,end_station_name
South Van Ness at Market,South Van Ness at Market
San Jose City Hall,San Jose City Hall
Mountain View City Hall,Mountain View City Hall
San Jose City Hall,San Jose City Hall
South Van Ness at Market,Market at 10th
Golden Gate at Polk,Golden Gate at Polk
Santa Clara at Almaden,Adobe on Almaden
San Salvador at 1st,San Salvador at 1st
South Van Ness at Market,South Van Ness at Market
/home/x-amicco


## Question 4

In [33]:
#creating the new file containing the pattern Van Ness

In [36]:
%%bash

grep "Van Ness" /anvil/projects/tdm/data/bay_area_bike_share/kaggle/trip.csv >$HOME/vanness.csv

In [37]:
#checking to make sure the vanness.csv file populated correctly

In [39]:
%%bash
cd $HOME
head vanness.csv
pwd

4576,63,8/29/2013 14:13,South Van Ness at Market,66,8/29/2013 14:14,South Van Ness at Market,66,520,Subscriber,94127
4299,83,8/29/2013 12:02,South Van Ness at Market,66,8/29/2013 12:04,Market at 10th,67,319,Subscriber,94103
4760,113,8/29/2013 17:01,South Van Ness at Market,66,8/29/2013 17:03,South Van Ness at Market,66,553,Subscriber,94103
5070,168,8/29/2013 21:43,South Van Ness at Market,66,8/29/2013 21:46,South Van Ness at Market,66,598,Subscriber,94115
4625,175,8/29/2013 15:02,San Francisco City Hall,58,8/29/2013 15:05,South Van Ness at Market,66,267,Subscriber,94115
4721,184,8/29/2013 16:27,Market at 10th,67,8/29/2013 16:30,South Van Ness at Market,66,416,Subscriber,94107
4720,188,8/29/2013 16:27,Market at 10th,67,8/29/2013 16:30,South Van Ness at Market,66,379,Subscriber,94109
4824,207,8/29/2013 17:38,Market at 10th,67,8/29/2013 17:42,South Van Ness at Market,66,632,Subscriber,94103
4765,228,8/29/2013 17:05,South Van Ness at Market,66,8/29/2013 17:08,Market at 10th,67,553,Subscrib

## Question 5

In [41]:
# seeing the head of the status.csv data 

In [50]:
%%bash
head -n6 /anvil/projects/tdm/data/bay_area_bike_share/kaggle/status.csv

station_id,bikes_available,docks_available,time
2,2,25,2013/08/29 12:06:01
2,2,25,2013/08/29 12:07:01
2,2,25,2013/08/29 12:08:01
2,2,25,2013/08/29 12:09:01
2,2,25,2013/08/29 12:10:01


In [45]:
#getting all the rows with data from station 2

In [76]:
%%bash
grep "^2," /anvil/projects/tdm/data/bay_area_bike_share/kaggle/status.csv > $HOME/station2.csv

In [77]:
#checking to make sure it populated correctly

In [78]:
%%bash
head $HOME/station2.csv
tail $HOME/station2.csv

2,2,25,2013/08/29 12:06:01
2,2,25,2013/08/29 12:07:01
2,2,25,2013/08/29 12:08:01
2,2,25,2013/08/29 12:09:01
2,2,25,2013/08/29 12:10:01
2,2,25,2013/08/29 12:11:01
2,2,25,2013/08/29 12:12:01
2,2,25,2013/08/29 12:13:01
2,2,25,2013/08/29 12:15:01
2,2,25,2013/08/29 12:16:02
2,18,9,2015-08-31 23:50:02
2,18,9,2015-08-31 23:51:02
2,18,9,2015-08-31 23:52:02
2,18,9,2015-08-31 23:53:03
2,18,9,2015-08-31 23:54:02
2,18,9,2015-08-31 23:55:02
2,18,9,2015-08-31 23:56:01
2,18,9,2015-08-31 23:57:02
2,18,9,2015-08-31 23:58:02
2,18,9,2015-08-31 23:59:02


In [79]:
#finding the number of rows in the station2.csv file

In [80]:
%%bash
tail -n +2 $HOME/station2.csv | wc -l

1046897


In [81]:
#getting all the rows with data from station 3

In [82]:
%%bash
grep "^3," /anvil/projects/tdm/data/bay_area_bike_share/kaggle/status.csv > $HOME/station3.csv

In [83]:
#checking to make sure it populated correctly

In [84]:
%%bash
head $HOME/station3.csv
tail $HOME/station3.csv

3,8,7,2013/08/29 12:06:01
3,8,7,2013/08/29 12:07:01
3,8,7,2013/08/29 12:08:01
3,8,7,2013/08/29 12:09:01
3,8,7,2013/08/29 12:10:01
3,8,7,2013/08/29 12:11:01
3,8,7,2013/08/29 12:12:01
3,8,7,2013/08/29 12:13:01
3,8,7,2013/08/29 12:15:01
3,8,7,2013/08/29 12:16:02
3,7,8,2015-08-31 23:50:02
3,7,8,2015-08-31 23:51:02
3,7,8,2015-08-31 23:52:02
3,7,8,2015-08-31 23:53:03
3,7,8,2015-08-31 23:54:02
3,7,8,2015-08-31 23:55:02
3,7,8,2015-08-31 23:56:01
3,7,8,2015-08-31 23:57:02
3,7,8,2015-08-31 23:58:02
3,7,8,2015-08-31 23:59:02


In [86]:
#finding the number of rows in the station3.csv file

In [87]:
%%bash
tail -n +2 $HOME/station3.csv | wc -l

1047112


In [69]:
#getting all the rows with data from station 4

In [74]:
%%bash
grep "^4," /anvil/projects/tdm/data/bay_area_bike_share/kaggle/status.csv > $HOME/station4.csv

In [88]:
#checking to make sure it populated correctly

In [89]:
%%bash
head $HOME/station4.csv
tail $HOME/station4.csv

4,0,11,2013/08/29 12:06:01
4,0,11,2013/08/29 12:07:01
4,0,11,2013/08/29 12:08:01
4,0,11,2013/08/29 12:09:01
4,0,11,2013/08/29 12:10:01
4,0,11,2013/08/29 12:11:01
4,0,11,2013/08/29 12:12:01
4,0,11,2013/08/29 12:13:01
4,0,11,2013/08/29 12:15:01
4,0,11,2013/08/29 12:16:02
4,1,10,2015-08-31 23:50:02
4,1,10,2015-08-31 23:51:02
4,1,10,2015-08-31 23:52:02
4,1,10,2015-08-31 23:53:03
4,1,10,2015-08-31 23:54:02
4,1,10,2015-08-31 23:55:02
4,1,10,2015-08-31 23:56:01
4,1,10,2015-08-31 23:57:02
4,1,10,2015-08-31 23:58:02
4,1,10,2015-08-31 23:59:02


In [91]:
#finding the number of rows in the station4.csv file

In [92]:
%%bash
tail -n +2 $HOME/station4.csv | wc -l

1047099


In [66]:
#getting all the rows with data from station 5

In [93]:
%%bash
grep "^5," /anvil/projects/tdm/data/bay_area_bike_share/kaggle/status.csv > $HOME/station5.csv

In [94]:
#checking to make sure it populated correctly

In [96]:
%%bash
head $HOME/station5.csv
tail $HOME/station5.csv

5,10,9,2013/08/29 12:06:01
5,10,9,2013/08/29 12:07:01
5,10,9,2013/08/29 12:08:01
5,10,9,2013/08/29 12:09:01
5,10,9,2013/08/29 12:10:01
5,10,9,2013/08/29 12:11:01
5,10,9,2013/08/29 12:12:01
5,10,9,2013/08/29 12:13:01
5,10,9,2013/08/29 12:15:01
5,10,9,2013/08/29 12:16:02
5,10,9,2015-08-31 23:50:02
5,10,9,2015-08-31 23:51:02
5,10,9,2015-08-31 23:52:02
5,10,9,2015-08-31 23:53:03
5,10,9,2015-08-31 23:54:02
5,10,9,2015-08-31 23:55:02
5,10,9,2015-08-31 23:56:01
5,10,9,2015-08-31 23:57:02
5,10,9,2015-08-31 23:58:02
5,10,9,2015-08-31 23:59:02


In [97]:
#finding the number of rows in the station5.csv file

In [98]:
%%bash
tail -n +2 $HOME/station5.csv | wc -l

1047141


A) There are 1046897 rows of data that correspond to Station 2

B) There are 1047112 rows of data that correspond to Station 3

C) There are 1047099 rows of data that correspond to Station 4

D) There are 1047141 rows of data that correspond to Station 5

## Pledge

By submitting this work I hereby pledge that this is my own, personal work. I've acknowledged in the designated place at the top of this file all sources that I used to complete said work, including but not limited to: online resources, books, and electronic communications. I've noted all collaboration with fellow students and/or TA's. I did not copy or plagiarize another's work.

> As a Boilermaker pursuing academic excellence, I pledge to be honest and true in all that I do. Accountable together – We are Purdue.